samply/bridgehead
# Moved all possible Environment Settings to specialised Docker Images

### What is included
- ID-Management will now properly start up
- Removed redundant environment variable settings (e.g. in internal.env, site.conf.dev)
- All settings required to be locally (e.g. apikeys and passwords), are now extracted from configurations that are pushed to the git repositories
- fixed most of the links in connector startup check of the connector (HOST must have other value than "localhost")
- proxy configuration should now be unified
- Prepared separate images for connector, patientlist and id-manager that set some ccp, gbn and c4 specific variables. See References for the implementations of those
### References
- [the source of this branch](https://github.com/samply/bridgehead/tree/refactor/preconfiguredImages)
- [the adjusted version of the teststandort site config for this branch](https://code.mitro.dkfz.de/projects/BD/repos/teststandort/browse?at=refs%2Fheads%2Frefactor%2FpreconfiguredImages)
- [the source for the specified share-client image](https://github.com/samply/share-client/tree/feature/environmentPreconfigurationTorben)
- [the source for the specified patientlist image](https://code.mitro.dkfz.de/projects/DEP/repos/bridgehead-mainzelliste-builder/browse?at=refs%2Fheads%2Fdevelop)
- [the source for the specified id-manager image](https://code.mitro.dkfz.de/projects/DEP/repos/bridgehead-magicpl-builder/browse?at=refs%2Fheads%2Fdevelop)
### How to test
First you need to prepare your local copy of the repository. Ensure that you fullfill following requirements:
- you have the bridgehead repository checked out with branch "refactor/preconfiguredImages"
- you have a local copy of [the teststandort site config from refactor/preconfiguredImages](https://code.mitro.dkfz.de/projects/BD/repos/teststandort/browse?at=refs%2Fheads%2Frefactor%2FpreconfiguredImages) in the folder site-config.
You can start a single bridgehead with the command
`docker-compose up -f dktk/docker-compose.yml --env-file site-config/dktk.env  up -d`
> **NOTE:** "dktk" is replaceable with "c4" and "gbn"

If you want to execute all bridgeheads on the same system, you need to add `-p bridgehead` to the docker-compose call.
This will allow you to start all bridgeheads one after another.
